packages/react-components/react-combobox/src/components/Option/useOption.tsx

```diff
@@ -34,21 +35,41 @@ export const useOption_unstable = (props: OptionProps, ref: React.Ref<HTMLElemen
   const optionRef = React.useRef<HTMLElement>(null);
   const optionValue = getValueString(value, props.children);
 
+  // use the id if provided, otherwise use a generated id
+  const defaultId = useId('fluent-option');
+  const id = React.useMemo(() => {
+    return props.id || defaultId;
+  }, [props.id, defaultId]);
```

Comment:
`useId` already has support for a default ID:
```suggestion
  const id = useId('fluent-option', props.id);
```